crates/scsi/src/backend.rs

```diff
@@ -0,0 +1,159 @@
+// SPDX-License-Identifier: Apache-2.0 or BSD-3-Clause
+
+use std::{
+    fs::File,
+    path::PathBuf,
+    sync::{Arc, RwLock},
+};
+
+use crate::scsi::emulation::{
+    block_device::{BlockDevice, FileBackend, MediumRotationRate},
+    target::EmulatedTarget,
+};
+use clap::Parser;
+use log::{error, info, warn};
+use thiserror::Error as ThisError;
+use vhost::vhost_user::{self, Listener};
+use vhost_user_backend::VhostUserDaemon;
+use vm_memory::{GuestMemoryAtomic, GuestMemoryMmap};
+
+use crate::vhu_scsi::VhostUserScsiBackend;
+
+#[derive(Debug, ThisError)]
+pub enum Error {
+    #[error("More than 256 LUNs aren't currently supported")]
+    TooManyLUNs,
+    #[error("Failed creating listener: {0}")]
+    FailedCreatingListener(vhost_user::Error),
+}
+
+pub type Result<T> = std::result::Result<T, Error>;
+
+#[derive(Parser)]
+struct ScsiArgs {
+    /// Make the images read-only.
+    ///
+    /// Currently, we don't actually support writes, but sometimes we want to
+    /// pretend the disk is writable to work around issues with some tools that
+    /// use the Linux SCSI generic API.
+    #[arg(long = "read-only", short = 'r')]
+    read_only: bool,
+    /// Tell the guest this disk is non-rotational.
+    ///
+    /// Affects some heuristics in Linux around, for example, scheduling.
+    #[arg(long = "solid-state")]
+    solid_state: bool,
+    /// Location of vhost-user socket.
+    #[clap(short, long)]
+    socket_path: PathBuf,
+    /// Images against which the SCSI actions are emulated.
+    images: Vec<PathBuf>,
+}
+
+fn create_backend(args: &ScsiArgs) -> Result<VhostUserScsiBackend> {
+    let mut backend = VhostUserScsiBackend::new();
+    let mut target = EmulatedTarget::new();
+    if args.images.len() > 256 {
+        error!("");
+        // This is fairly simple to add; it's just a matter of supporting the right LUN
+        // encoding formats.
+        return Err(Error::TooManyLUNs);
+    }
+    if !args.read_only {
```

Comment:
I would add blank lines after every block of code, like the if block above and this one.